google/cloud/bigtable/examples/table_admin_snippets.cc

```diff
@@ -657,44 +658,49 @@ void DropRowsByPrefix(
   (std::move(admin), argv.at(0), argv.at(1), argv.at(2), argv.at(3));
 }
 
-// TODO(#7732) - update this sample to use the helper method
 void WaitForConsistencyCheck(
-    google::cloud::bigtable_admin::BigtableTableAdminClient const&,
+    google::cloud::bigtable_admin::BigtableTableAdminClient admin,
     std::vector<std::string> const& argv) {
-  auto old_admin = google::cloud::bigtable::TableAdmin(
-      google::cloud::bigtable::MakeAdminClient(argv.at(0)), argv.at(1));
-
   //! [wait for consistency check]
   namespace cbt = ::google::cloud::bigtable;
+  namespace cbta = ::google::cloud::bigtable_admin;
+  using ::google::cloud::CompletionQueue;
   using ::google::cloud::future;
+  using ::google::cloud::Status;
   using ::google::cloud::StatusOr;
-  [](cbt::TableAdmin admin, std::string const& table_id) {
-    StatusOr<std::string> consistency_token =
-        admin.GenerateConsistencyToken(table_id);
-    if (!consistency_token) {
-      throw std::runtime_error(consistency_token.status().message());
+  [](cbta::BigtableTableAdminClient admin, std::string const& project_id,
+     std::string const& instance_id, std::string const& table_id) {
+    std::string table_name = cbt::TableName(project_id, instance_id, table_id);
+
+    // First, generate the consistency token
+    StatusOr<google::bigtable::admin::v2::GenerateConsistencyTokenResponse>
+        result = admin.GenerateConsistencyToken(table_name);
+    if (!result) {
+      throw std::runtime_error(result.status().message());
     }
-    future<StatusOr<cbt::Consistency>> consistent_future =
-        admin.WaitForConsistency(table_id, *consistency_token);
-    future<void> fut = consistent_future.then(
-        [&consistency_token](future<StatusOr<cbt::Consistency>> f) {
-          auto is_consistent = f.get();
-          if (!is_consistent) {
-            throw std::runtime_error(is_consistent.status().message());
-          }
-          if (*is_consistent == cbt::Consistency::kConsistent) {
-            std::cout << "Table is consistent with token " << *consistency_token
-                      << "\n";
-          } else {
-            std::cout
-                << "Table is not yet consistent, Please try again later with"
-                << " the same token (" << *consistency_token << ")\n";
-          }
-        });
-    fut.get();  // simplify example by blocking until operation is done.
+    std::string consistency_token = result->consistency_token();
+
+    // Spin off a thread to run the operation. An application will likely want
```

Comment:
Ugh... Okay, I see why this works now, you have a completely different thread pool for `AsyncWaitForConsistency`.  That is ... not desirable, spinning one of these up for each request would be expensive, ideally an application should have a single completion queue and thread pool.

I hate that you are blocked on this, but the API for `AsyncWaitForConsistency()` is not something I would like us to expose to customers (even if none of them uses it).

Reply:
Maybe if `AsyncWaitForConsistency()` used the internal CQ, maybe with an accessor in `internal::` or some kind of `friend` function or struct.